mongo/integration/client_test.go

```diff
@@ -600,6 +603,99 @@ func TestClient(t *testing.T) {
 		assert.Equal(t, 0, closed, "expected no connections to be closed")
 	})
 
+	mt.Run("RTT90 is monitored", func(mt *mtest.T) {
+		if testing.Short() {
+			t.Skip("skipping integration test in short mode")
+		}
+
+		// Reset the client with a dialer that delays all network round trips by 300ms and set the
+		// heartbeat interval to 100ms to reduce the time it takes to collect RTT samples.
+		mt.ResetClient(options.Client().
+			SetDialer(newSlowConnDialer(slowConnDialerDelay)).
+			SetHeartbeatInterval(reducedHeartbeatInterval))
+
+		// Assert that RTT90s are eventually >300ms.
+		topo := getTopologyFromClient(mt.Client)
+		assert.Soon(mt, func() {
+			for {
+				time.Sleep(100 * time.Millisecond)
+
+				// Wait for all of the server's RTT90s to be >300ms.
+				done := true
+				for _, desc := range topo.Description().Servers {
+					server, err := topo.FindServer(desc)
+					assert.Nil(mt, err, "FindServer error: %v", err)
+					if server.RTT90() <= 300*time.Millisecond {
+						done = false
+					}
+				}
+				if done {
+					return
+				}
```

Comment:
What if this condition never becomes true? We need a way to time out this goroutine to prevent it from running indefinitely.